sympy/functions/combinatorial/factorials.py

```diff
@@ -357,6 +359,10 @@ class factorial2(CombinatorialFunction):
     @classmethod
     def eval(cls, arg):
         # TODO: extend this to complex numbers?
+        check = arg.is_real
```

Comment:
This does not cover real non-integers. It's strange to define double factorial for non-real arguments but leave real non-integers undefined.

Reply:
Also, at this point arg could be symbolic, as we are not within `is_Number` branch.  So, double factorial would be rewritten for expressions like x + I with symbolic real x. But if it can't be evaluated, so why the rewrite in terms of gamma?